release.sh

```diff
@@ -98,95 +109,119 @@ if [[ "$CONFIRM" != "y" ]]; then
     exit 1
 fi
 
-# Make a new detached HEAD
 git checkout "$BRANCH"
 git pull
-git checkout --detach
-
-# Edit package.json to the right version (see
-# http://stackoverflow.com/a/22084103 for why we need the .bak file to make
-# this mac & linux compatible)
-sed -i.bak -E 's|"version": "[^"]+",|"version": "'$VERSION'",|' package.json
-rm -f package.json.bak
-
-# Build generated files and add them to the repository
-git clean -fdx dist
-yarn dist
-sed -i.bak -E '/^\/dist\/$/d' .gitignore
-rm -f .gitignore.bak
-git add .gitignore dist/
-
-# Edit docs to use CSS from CDN
-grep -l '{@stylesheet: katex.min.css}' docs/*.md | xargs sed -i.bak \
-    's|{@stylesheet: katex.min.css}|<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@./dist/katex.min.css" integrity="sha384-katex.min.css" crossorigin="anonymous"/>|'
-
-# Update the version number in CDN URLs included in the README and the documentation,
-# and regenerate the Subresource Integrity hash for these files.
-node update-sri.js "${VERSION}" README.md contrib/*/README.md dist/README.md \
-    docs/*.md docs/*.md.bak website/pages/index.html
-
-# Generate a new version of the docs and publish the website
-pushd website
-yarn run version "${VERSION}"
-
-# Restore docs to use local built CSS
-for file in ../docs/*.md.bak; do
-    mv -f "$file" "${file%.bak}"
-done
 
-USE_SSH=true yarn publish-gh-pages
-popd
+if [[ $BRANCH != "master" ]]; then
+    # Build generated files
+    yarn build
+
+    # Regenerate the Subresource Integrity hash in the README and the documentation
+    node update-sri.js "${VERSION}" README.md contrib/*/README.md \
+        docs/*.md website/pages/index.html website/versioned_docs/version-$VERSION/*.md
+
+    # Make the commit and push
+    git add README.md contrib/*/README.md \
+        docs website/pages/index.html website/versioned_docs/
```

Comment:
I wonder if it's worth checking for any unstaged changes or new files after this command.

Reply:
@kevinbarabash I move the check to the top level.